docs/kb/band.md

```diff
@@ -3,105 +3,185 @@ order: 12
-To scaffold an oracle:
+The basic syntax to scaffold a band oracle module is:
 
 ```bash
 starport scaffold band [queryName] --module [moduleName]
 ```
 
-### Acknowledgement
+Customize your band oracle with flags:
 
-The BandChain oracle will return the ack messages with the request's id, and we save the last request id for future queries.
+- --module string - name of the new IBC Module to add the packets to
+- --path string - file path of the app, efault is the current directory (`"."`)
+- --signer string - signer label, default is `creator`
 
-## Files and Directories
+### Acknowledgement messages
 
-When you scaffold a BandChain oracle, the following files and directories are created and modified:
+The BandChain oracle returns the ack messages with the request id. The last request id is saved for future queries.
+
+## Files and directories
+
+When you scaffold a BandChain oracle module, the following files and directories are created and modified:
 
 - `proto`: oracle request and response data.
 - `x/module_name/keeper`: IBC hooks, gRPC message server.
 - `x/module_name/types`: message types, IBC events.
 - `x/module_name/client/cli`: CLI command to broadcast a transaction containing a message with a packet.
 - `x/module_name/oracle.go`: BandChain oracle packet handlers.
 
-## BandChain Oracle Scaffold Example
+## Scaffold a BandChain oracle chain
 
-The following command scaffolds the IBC-enabled oracle. by default, the starport scaffold oracle for [coin rates](https://laozi-testnet4.cosmoscan.io/oracle-script/37#bridge) request and result.
+First, scaffold a chain but don't scaffold a default module:
 
-```shell
-$ starport scaffold chain github.com/cosmonaut/oracle --no-module && cd oracle 
-$ starport scaffold module consuming --ibc
-$ starport scaffold band coinRates --module consuming
+```bash
+starport scaffold chain github.com/cosmonaut/oracle --no-module 
+```
+
+Next, change to the new `oracle` directory and scaffold an IBC-enabled module named `consuming`: 
+
+```bash
+cd oracle 
+starport scaffold module consuming --ibc
+```
+
+Finally, scaffold a BandChain query oracle that can request real-time data:
+
+```
+starport scaffold band coinRates --module consuming
 ```
 
-Note: BandChain module uses version "bandchain-1". Make sure to update the `keys.go` file accordingly.
+So far, you have scaffolded:
+
+- A new `oracle` chain without a default module
+- A new IBC-enabled `consuming` module
+- A new `coinRates` BandChain query oracle
+
+Now it's time to change the data.
 
-`x/oracle/types/keys.go`
+## Update version
+
+The output of the `starport scaffold band coinRates --module consuming` command prompts you to update the `keys.go` file.
+
+In the `x/oracle/types/keys.go` file, update the `Version` variable in the `const` block to the required version that the IBC module supports:
 
 ```go
-const Version = "bandchain-1"
+const (
+    
+    // Version defines the current version the IBC module supports
+    Version = "bandchain=1"
```

Comment:
@Pantani I originally used the hyphen as documented in

but when I ran `starport relayer connect `

I received this error:

`Error when broadcasting tx C7859079EBB2B9B5BDF963864F26F7244EF2895DE91BC08A99D20371876933F3 at height 679. Code: 1501; Raw log: failed to execute message; message index: 0: channel open try callback failed: got: bandchain-1, expected bandchain=1: invalid version
`

I deleted the original chain, followed all steps again, and still got the error. Can you help me walk through the steps?